### PR TITLE
Add a warning to the `detect_anomaly` flag

### LINKED ISSUE #17359: Add a performance warning to pl.Trainer parameter detect_anomaly.
### Description & Motivation

This detect_anomaly parameter greatly affects performance and is usually only used in debugging.  
Sometimes after debugging, user forget setting detect_anomaly to False, resulting in a performance loss during training.   
Hope add a warning to notice user that detect_anomaly flag will cause performance lose.  

In my experiment, the train speed when detect_anomaly is turned off is 2.5x when it is turned on.

### Pitch

Add a performance warning when detect_anomaly enabled.

### Alternatives

_No response_

### Additional context

_No response_

cc @borda @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by One-sixth**
## What does this PR do?

Fixes #17359

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: src/lightning/pytorch/trainer/trainer.py [hunk 1]
```diff
@@ -454,6 +454,10 @@ def __init__(
             GradClipAlgorithmType(gradient_clip_algorithm.lower()) if gradient_clip_algorithm is not None else None
         )
 
+        if detect_anomaly:
+            rank_zero_info(
+                "You have turned on detect_anomaly flag, which can significantly slow down compute speed and is recommended only for model debugging."
```
Comment (carmocca) [inline src/lightning/pytorch/trainer/trainer.py:459]:
```suggestion
                "You have turned on `Trainer(detect_anomaly=True)`. This will significantly slow down compute speed and is recommended only for model debugging."
```

**Review by carmocca — APPROVED** [review]
Thanks!

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]
